modal/_utils/local_file_filter.py

```diff
@@ -0,0 +1,54 @@
+# Copyright Modal Labs 2024
+from pathlib import Path
+from typing import Callable
+
+from .pattern_matcher import PatternMatcher
+
+
+class LocalFileFilter:
```

Comment:
Maybe this should be called "IgnorePatterns" or something, and make it clear in that it *excludes* on `True` values.  The word "Filter" and some of these docstrings make it sound like its the other way around

Reply:
Hm, this class in itself doesn't exclude or include specifically though, I guess that's more how it's used in the `Image` methods, but I'm open to renaming it, but maybe something like `FilePathPatternMatcher` or something to show that it's related to our `PatternMatcher`?

Reply:
I'm still a little confused about we need to be wrapping our own `PatternMatcher` object here

Reply:
Sorry, you're right, not sure what I was thinking in the original comment. This returns Truthy for matching items and then it makes sense that it's called a Matcher or Filter or something like this

Reply:
Went for using the `PatternMatcher` object directly but modifying it to take in `Path` objects and add an invert operator